app/code/BVMyBlog/Blog/Controller/Adminhtml/Post/Save.php

```diff
@@ -0,0 +1,120 @@
+<?php
+declare(strict_types=1);
+
+namespace BVMyBlog\Blog\Controller\Adminhtml\Post;
+
+use BVMyBlog\Blog\Api\BlogRepositoryInterface;
+use BVMyBlog\Blog\Model\Blog;
+use BVMyBlog\Blog\Model\BlogFactory;
+use Magento\Framework\App\Action\Action;
+use Magento\Framework\App\Action\Context;
+use Magento\Framework\App\Action\HttpPostActionInterface;
+use Magento\Framework\App\Request\DataPersistorInterface;
+use Magento\Framework\Controller\ResultInterface;
+use Magento\Framework\Exception\CouldNotSaveException;
+use Magento\Framework\Exception\NoSuchEntityException;
+
+/**
+ * Saves data from form
+ */
+class Save extends Action implements HttpPostActionInterface
+{
+    const ADMIN_RESOURCE = 'BVMyBlog_Blog::blog_manage_posts';
+
+    /**
+     * @var mixed $blogFactory
+     */
+    private $blogFactory;
+
+    /**
+     * @var DataPersistorInterface
+     */
+    protected $dataPersistor;
+
+    /**
+     * @var BlogRepositoryInterface
+     */
+    private $blogRepository;
+
+    /**
+     * @param Context $context
+     * @param DataPersistorInterface $dataPersistor
+     * @param BlogFactory $blogFactory
+     * @param BlogRepositoryInterface $blogRepository
+     */
+    public function __construct(
+        Context $context,
+        DataPersistorInterface $dataPersistor,
+        BlogFactory $blogFactory,
+        BlogRepositoryInterface $blogRepository
+    ) {
+        $this->dataPersistor = $dataPersistor;
+        $this->blogFactory = $blogFactory;
+        $this->blogRepository = $blogRepository;
+        parent::__construct($context);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function execute()
+    {
+        $resultRedirect = $this->resultRedirectFactory->create();
+        $data = $this->getRequest()->getPostValue();
+
+        if ($data) {
```

Comment:
$id = $this->getRequest()->getParam('id')
if (empty($data)) {
    $this->messageManager->addErrorMessage(__('Please specify valid data.'));
    return $resultRedirect->setPath('*/*/edit', ['post_id' => $id]);
}